### DISCUSSION TITLE
keys in CollectionSpec.Programs may differ from those in Collection.Programs

### DISCUSSION THREAD

**Discussion by christian-2** [cilium/ebpf#1330] [Q&A] [answered]
I have noticed that the keys in `Collection.Programs` are apparently limited to 15 characters, whereas the keys in `CollectionSpec.Programs` may be arbitrary strings. So corresponding pairs of keys may not be equal but a prefix relationship may hold instead.

Should the source code documentation perhaps be amended accordingly (if I diagnosis is correct)? If so, I could prepare a PR.

**Comment by brycekahle** [comment]
The key used in the `Collection.Programs` map comes from the key in `CollectionSpec.Programs`: https://github.com/cilium/ebpf/blob/c2563e4e6981ff1c7247582d7f1970cc733338b6/collection.go#L385

**Reply by brycekahle** [reply, marked as answer]
The kernel only supports passing 15 characters for a name when loading the eBPF program, but this library doesn't hold onto that value. It is only created and used during the syscall.
